src/test/java/ch/puzzle/eft/controller/SiteControllerTest.java

```diff
@@ -94,4 +107,34 @@ void shouldNotAcceptExamNumberWithNoMatchingFiles() throws Exception {
                     .andExpect(model().attributeDoesNotExist("examFiles"));
     }
 
-}
+    @Test
+    void shouldCreateCookie() throws Exception {
+        MvcResult mvcResult = this.mockMvc.perform(post("/").with(csrf()))
+                                          .andExpect(status().isFound())
+                                          .andExpect(header().string(HttpHeaders.LOCATION, "/"))
+                                          .andReturn();
+        String cookieHeader = mvcResult.getResponse()
+                                       .getHeader(HttpHeaders.SET_COOKIE);
+        Assertions.assertNotNull(cookieHeader);
+        Assertions.assertTrue(cookieHeader.contains("SameSite=Strict"));
+        Assertions.assertTrue(cookieHeader.contains("cookie-consent=true;"));
+        Assertions.assertTrue(cookieHeader.contains("Max-Age=31536000;"));
+        Assertions.assertTrue(cookieHeader.contains("HttpOnly;"));
```

Comment:
check if https works

Reply:
HttpOnly is no longer set so i removed the line

Reply:
Https works per defulte and the HttpOnly Flag dose not inpact that.

There is a Secure flag  to make it only Https but there is no other way around so i cant write a test that makes sure Https works unless i make it only Https accessable